app/social_transport.py

```
@@ -0,0 +1,145 @@
+        policy = CapabilityPolicy(frozenset(manifest.capabilities))
+        token = broker.access_token(credential)
+        return cls(access_token=token, capabilities=policy, base_url=base_url, timeout=timeout)
+
+    def _get_json(self, path: str, *, fields: tuple[str, ...]) -> dict[str, Any]:
+        if not path.startswith("/") or ".." in path or "?" in path or "#" in path:
+            raise ValueError("unsafe Instagram API path")
+        query = urlencode({"fields": ",".join(fields)}) if fields else ""
+        url = f"{self._base_url}{path}" + (f"?{query}" if query else "")
+        request = Request(url, method="GET", headers={"Authorization": f"Bearer {self._access_token}", "Accept": "application/json", "User-Agent": "MatVerse-Governed-Bridge/1.0"})
+        try:
+            with urlopen(request, timeout=self._timeout) as response:  # nosec B310
```

Ruff (0.16.3)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)